### PR TITLE
Spike/32415 change dfenumber joins to schoolid

### PR DESCRIPTION
**Pull request by danwilliams**
Found all queries using `dfeNumber` to join on the school table. Changed them all to instead use schoolID, which is already available in the user's session, to remove the need for all the joins.

### CODE REVIEW

File: admin/services/access-arrangements.service.js [hunk 1]
```diff
@@ -19,18 +19,18 @@ accessArrangementsService.getAccessArrangements = async () => {
 /**
  * Submit access arrangements for single pupil
  * @param {Object} submittedData
- * @param {Number} dfeNumber
+ * @param {Number} schoolID
```
Comment (GeorgeChatzigiannis) [inline admin/services/access-arrangements.service.js]: @danwilliams that should be `schoolId` on all references

Comment (danwilliams) [reply admin/services/access-arrangements.service.js]: ...it is; sorry, I think I am missing something about the comment?

Comment (GeorgeChatzigiannis) [reply admin/services/access-arrangements.service.js]: @danwilliams D shouldn't be capital.
`schoolID` -> `schoolId`

Comment (danwilliams) [reply admin/services/access-arrangements.service.js]: Well-spotted :)

File: admin/services/data-access/pupil.data.service.js [hunk 2]
```diff
@@ -241,27 +239,26 @@ pupilDataService.sqlCreate = async (data) => {
...
     ON g.pupil_id = p.id
   WHERE p.pin IS NOT NULL
-  AND s.dfeNumber = @dfeNumber
+  AND p.school_id = @schoolID
   AND p.pinExpiresAt IS NOT NULL
   AND p.pinExpiresAt > GETUTCDATE()
   ORDER BY p.lastName ASC, p.foreName ASC, p.middleNames ASC, dateOfBirth ASC
   `
-  return sqlService.query(sql, [paramDfeNumber])
+  console.log('TEST')
```
Comment (GuyHarwood) [inline admin/services/data-access/pupil.data.service.js]: remove console.log

File: admin/services/data-access/pupil.data.service.js [hunk 3]
```diff
@@ -241,27 +239,26 @@ pupilDataService.sqlCreate = async (data) => {
 /**
  * Find pupils for a school with pins that have not yet expired, for
  * a specific pin environment (live / fam)
- * @param dfeNumber
+ * @param schoolID
```
Comment (GuyHarwood) [inline admin/services/data-access/pupil.data.service.js]: change capitalised ID

File: admin/services/data-access/pupil.data.service.js [hunk 4]
```diff
@@ -10,22 +10,21 @@ const sqlService = require('./sql.service')
 /** SQL METHODS */
 
 /**
- * Fetch all pupils for a school by dfeNumber sorted by specific column.
- * @param dfeNumber
+ * Fetch all pupils for a school by schoolID sorted by specific column.
+ * @param schoolID
```
Comment (GuyHarwood) [inline admin/services/data-access/pupil.data.service.js]: change capitalisation of ID

File: admin/services/restart.service.js [hunk 5]
```diff
@@ -20,13 +20,13 @@ restartService.totalChecksAllowed = restartService.totalRestartsAllowed + 1
 
 /**
  * Get pupils who are eligible for restart
- * @param dfeNumber
+ * @param schoolD
```
Comment (GuyHarwood) [inline admin/services/restart.service.js]: rename to Id

**Comment by GuyHarwood** [comment]
@GeorgeChatzigiannis please fix conflicts and changes requested

**Comment by GeorgeChatzigiannis** [comment]
@GuyHarwood @jon-shipley please review this when possible

**Comment by GeorgeChatzigiannis** [comment]
@jon-shipley @GuyHarwood please review the last commit: c7a7df57853c1167c62257d06e37749fb6cf0ffd

File: admin/services/data-access/pupils-not-taking-check.data.service.js [hunk 6]
```diff
@@ -45,7 +45,9 @@ const pupilsNotTakingCheckDataService = {
       FROM ${sqlService.adminSchema}.[pupil] p 
       LEFT JOIN ${sqlService.adminSchema}.[pupilAttendance] pa ON p.id = pa.pupil_id AND pa.isDeleted=0
       LEFT JOIN ${sqlService.adminSchema}.[pupilGroup] pg ON p.id = pg.pupil_id
-      WHERE p.school_id = @schoolId AND pa.id IS NULL
+      WHERE p.school_id = @schoolId
+      AND p.pupilStatus_id = 1
```
Comment (jon-shipley) [inline admin/services/data-access/pupils-not-taking-check.data.service.js]: We shouldn't use IDs as we don't know what order they be inserted on a particular env.  Please replace with a subquery to fetch the id.

IDs could be safely memo'ised by the app if needed.

Comment (GeorgeChatzigiannis) [reply admin/services/data-access/pupils-not-taking-check.data.service.js]: @jon-shipley I raised a similar concern in a past PR and the change was not made for performance reasons.
I made the change here so the query is updated.

File: admin/services/data-access/pupils-not-taking-check.data.service.js [hunk 7]
```diff
@@ -42,11 +42,12 @@ const pupilsNotTakingCheckDataService = {
...
         p.urlSlug,
         pg.group_id
-      FROM ${sqlService.adminSchema}.[pupil] p 
+      FROM ${sqlService.adminSchema}.[pupil] p
+      JOIN ${sqlService.adminSchema}.pupilStatus ps ON ps.id = p.pupilStatus_id 
       LEFT JOIN ${sqlService.adminSchema}.[pupilAttendance] pa ON p.id = pa.pupil_id AND pa.isDeleted=0
       LEFT JOIN ${sqlService.adminSchema}.[pupilGroup] pg ON p.id = pg.pupil_id
       WHERE p.school_id = @schoolId
-      AND p.pupilStatus_id = 1
+      AND ps.code = 'UNALLOC'
```
Comment (jon-shipley) [inline admin/services/data-access/pupils-not-taking-check.data.service.js]: 💯

**Pull request merged**